hs_access_control/models/user.py

```diff
@@ -968,41 +1061,63 @@ def edit_resources(self):
         if not self.user.is_active:
             raise PermissionDenied("Requesting user is not active")
 
-        return BaseResource.objects.filter(Q(raccess__immutable=False) &
-                                           (Q(r2urp__user=self.user,
-                                              r2urp__privilege__lte=PrivilegeCodes.CHANGE) |
-                                            Q(r2grp__group__gaccess__active=True,
-                                                r2grp__group__g2ugp__user=self.user,
-                                              r2grp__privilege__lte=PrivilegeCodes.CHANGE)))\
-                                   .distinct()
+        # This is a mouthful.
+        # a resource is editable if
+        # 1. it's shared with the user and editable.
+        # 2. it's shared with a group that has edit privilege and contains the user,
+        # 3. it's shared with a group that has edit privilege and a community that
+        #    contains the user, and the community share preserves edit
+
+        return BaseResource.objects.filter(
+            # user has direct access
+            Q(raccess__immutable=False,
+              r2urp__user=self.user,
+              r2urp__privilege__lte=PrivilegeCodes.CHANGE) |
+            # user has direct access through being a member of a group
+            Q(raccess__immutable=False,
+              r2grp__group__gaccess__active=True,
+              r2grp__group__g2ugp__user=self.user,
+              r2grp__privilege=PrivilegeCodes.CHANGE) |
+            # user has access by being a member of a privileged group in the same community
+            # Note: CHANGE privilege overrides allow_view flag.
+            Q(raccess__immutable=False,
+              r2grp__group__gaccess__active=True,
```

Comment:
As before. Filter conditions `r2grp__group__gaccess__active=True` and `r2grp__privilege=PrivilegeCodes.CHANGE` probably not needed.

Reply:
All of these are needed.